### PR TITLE
[SIE-12] add Wizard Layout

### PR DESCRIPTION
**Pull request by alecherryy**

### CODE REVIEW

File: .storybook/main.js [hunk 1]
```diff
@@ -1,7 +1,8 @@
 module.exports = {
   "stories": [
     "../src/**/*.stories.mdx",
-    "../src/**/*.stories.@(js|jsx|ts|tsx)"
+    "../src/stories/components/**/*.stories.@(js|jsx|ts|tsx)",
+    "../src/stories/layouts/**/*.stories.@(js|jsx|ts|tsx)",
```
Comment (alecherryy) [inline .storybook/main.js:5]: Includes templates in the `layouts/` folder.

File: src/scss/_mixins.scss [hunk 2]
```diff
@@ -16,4 +17,34 @@
   &:focus {
     outline: none;
   }
+}
+
+// remove list default styles
+@mixin list-clean {
```
Comment (alecherryy) [inline src/scss/_mixins.scss:23]: New `@mixins` to style lists quickly.

File: src/stories/components/Constrain/Constrain.js [hunk 3]
```diff
@@ -8,20 +8,20 @@ import PropTypes from 'prop-types';
...
+ * @param {node} children Class modifiers of the component.
  * @return {object} (
  *   <Constrain modifierClasses={modifierClasses} />
  *      {content}
  *   </Constrain>
  * )
  */
 
-export const Constrain = ({ modifierClasses, content}) => {
+export const Constrain = ( { children, modifierClasses }) => {
```
Comment (alecherryy) [inline src/stories/components/Constrain/Constrain.js:19]: Changed this for usage consistency.

File: src/stories/components/Constrain/Constrain.stories.js [hunk 4]
```diff
@@ -12,69 +11,40 @@ export default {
...
-        iriure iracundia ea. Eam mazim efficiantur delicatissimi ei.
-        Duo at ornatus omnesque conclusionemque, ullum periculis in mei,
-        quo no sanctus eloquentiam.</p></div>,
 };
 
 // Wde Constrain
 export const Wide = Template.bind({});
 Wide.args = {
     modifierClasses: 'constrain--wide',
-    content: <div><h3>This is a Wide Constrain</h3>
```
Comment (alecherryy) [inline src/stories/components/Constrain/Constrain.stories.js:44]: Not sure what's the best way to display dummy content.

File: src/stories/layouts/Wizard/Wizard.js [hunk 5]
```diff
@@ -0,0 +1,106 @@
...
+                { SW && <Controls SW={SW} /> }
+            </Constrain>
+        );
+    }
+    return (
+        <div>
+            { children.length === labels.length ?
+                renderWizard() :
+                <Constrain modifierClasses="constrain--small">
+                    A message goes here.
```
Comment (alecherryy) [inline src/stories/layouts/Wizard/Wizard.js:60]: This is a placeholder for now. We should probably create an `<Alert />` component at some point to provide system feedback when needed.

**Comment by alecherryy** [comment]
To whomever reviews this, please run the branch on your local to make sure the build, components and styles render correctly and to confirm mobile behavior. Thanks!

File: src/stories/components/Steps/Steps.js [hunk 6]
```diff
@@ -0,0 +1,73 @@
+import './styles.scss';
+
+import React from 'react';
+import PropTypes from 'prop-types';
+
+/**
+ * Component for step prgress element.
```
Comment (fernandowinfield) [inline src/stories/components/Steps/Steps.js:7]: Teeny-tiny typo (sorry don't hit me)

File: src/stories/components/Steps/Steps.js [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+ *
+ * @component
+ * @param {number} props of the component.
+ * @param {array} labels of each step.
+ * @return {object} (
+ *   <Steps props={props
+ *      labels={labels} />
+ * )
```
Comment (fernandowinfield) [inline src/stories/components/Steps/Steps.js:15]: Just curious if the `number` parameter is kind of a generic param that can take the form of any of `totalSteps`, `currentStep`, and/or `goToStep`. I saw in the Steps.stories.js file you give all of those 3 as props of `<Steps/>`, so I was wondering if we could simplify the documentation and only have one @param, which would be called "props".

Maybe I am just overcomplicating things so just disregard this comment if that's the case.

**Review by fernandowinfield — COMMENTED** [review]
Left a few comments that are more suggestions than actual changes requests. The Wizard component looks really cool, by the way 🧙‍♂️.

One last thing: I checked out this pull request's branch locally and it all looks good in all screen sizes, but I wonder if we should add unit tests here before merging?

**Comment by alecherryy** [comment]
> Left a few comments that are more suggestions than actual changes requests. The Wizard component looks really cool, by the way 🧙‍♂️.
> 
> One last thing: I checked out this pull request's branch locally and it all looks good in all screen sizes, but I wonder if we should add unit tests here before merging?

I thought about that too. I think it's a good idea!

File: src/stories/components/Steps/Steps.js (hunk 7 above)
Comment (alecherryy) [reply src/stories/components/Steps/Steps.js:15]: I honestly wasn't sure if the param can be `prop` or it should be object or whatnot; I'll do some googling!

File: src/scss/_base.scss [hunk 8]
```diff
@@ -101,6 +101,7 @@ h6,
 %h6 {
   @extend %hN;
   font-size: $font-size-h6;
+  text-transform: uppercase;
```
Comment (cmutuc25) [inline src/scss/_base.scss:104]: This will make all `<h6>` uppercase right?

File: src/stories/layouts/Wizard/Wizard.js [hunk 9]
```diff
@@ -0,0 +1,106 @@
...
+    const onStepChange = (stats) => {
+        // might need later on
+    };
+
+    const setInstance = (SW) => updateState({
+        ...state,
+        SW,
+    });
+
+    const { SW } = state;
```
Comment (cmutuc25) [inline src/stories/layouts/Wizard/Wizard.js:36]: Can we change this to a more descriptive name?

File: src/stories/layouts/Wizard/styles.scss [hunk 10]
```diff
@@ -0,0 +1,71 @@
+@import '../../../scss/variables.scss';
+
+// @file
+// Wizard
+
+.wizard {
+  display: grid;
+  grid-template-columns: repeat(1, 1fr);
+  grid-gap: $spacing-5;
```
Comment (cmutuc25) [inline src/stories/layouts/Wizard/styles.scss:9]: `grid-gap` is deprecated, I think `gap` should be used here

**Review by cmutuc25 — CHANGES_REQUESTED** [review]

**Review by cmutuc25 — APPROVED** [review]

File: src/scss/_base.scss (hunk 8 above)
Comment (alecherryy) [reply src/scss/_base.scss:104]: Yes!

File: src/stories/layouts/Wizard/Wizard.js (hunk 9 above)
Comment (alecherryy) [reply src/stories/layouts/Wizard/Wizard.js:36]: Yes!

File: src/stories/layouts/Wizard/styles.scss (hunk 10 above)
Comment (alecherryy) [reply src/stories/layouts/Wizard/styles.scss]: Did not know that lol thanks!

**Pull request merged**